ArbFile.js

```diff
@@ -0,0 +1,346 @@
+        this.parsedData = JSON.parse(data);
+    }
+
+    for (var property in this.parsedData) {
+        if (property[0] !== '@') {
+            var r = this.API.newResource({
+                resType: "string",
+                project: this.project.getProjectId(),
+                key: ArbFile.unescapeString(property),
+                sourceLocale: this.project.sourceLocale,
+                source: ArbFile.cleanString(this.parsedData[property]),
+                autoKey: true,
+                pathName: this.pathName,
+                state: "new",
+                comment: undefined,
+                datatype: this.datatype,
+                index: this.resourceIndex++
+            });
+            this.set.add(r);
+        } else {
+            this.logger.debug("[" + property + "] string is not for the translation.");
+        }
+    }
+};
+
+/**
+ * Extract all the localizable strings from the arb file and add them to the
+ * project's translation set.
+ */
+ArbFile.prototype.extract = function() {
+    this.logger.debug("Extracting strings from " + this.pathName);
+    if (this.pathName) {
+        var p = path.join(this.project.root, this.pathName);
+        try {
+            var data = fs.readFileSync(p, "utf8");
+            if (data) {
+                this.parse(data);
+            }
+        } catch (e) {
+            this.logger.warn("Could not read file: " + p);
+        }
+    }
+};
+
+/**
+ * Return the set of resources found in the current arb file.
+ *
+ * @returns {TranslationSet} The set of resources found in the
+ * current arb file.
+ */
+ArbFile.prototype.getTranslationSet = function() {
+    return this.set;
+}
+
+// we don't localize or write arb source files
+ArbFile.prototype.write = function() {};
+
+/**
+ * Return the location on disk where the version of this file localized
+ * for the given locale should be written.
+ * @param {String] locale the locale spec for the target locale
+ * @returns {String} the localized path name
+ */
+ArbFile.prototype.getLocalizedPath = function(locale) {
+    var mapping = this.mapping || this.type.getMappings(this.pathName || "") || this.type.getDefaultMapping();
+    
+    var splitLocale = locale.split("-");
+    this.baseLocale = Utils.isBaseLocale(locale);
+    var resDir = this.project.getResourceDirs("arb")[0] || ".";
+    var lo = locale;
+
+    if (this.baseLocale) {
+        lo = splitLocale[0];
+    }
+    var path = this.API.utils.formatPath(mapping.template, {
+        sourcepath: this.pathName,
+        resourceDir: resDir,
+        locale: lo
+    });
+    return path;
+};
+
+ArbFile.prototype.getfullLocalizedPath = function(locale) {
+    var respath = this.getLocalizedPath(locale);
+    return path.join(this.project.target, respath);
+}
+
+ArbFile.prototype._checkEmptyObj = function(obj) {
+    return (Object.keys(obj).length > 0) ? false: true;
+}
+
+ArbFile.prototype._getLocaleField = function(locale) {
+    var splitLocale = locale.split("-");
```

Comment:
Shouldn't this use the Locale class from ilib?

Reply:
I updated it.